cmd/root.go

```diff
@@ -222,6 +229,116 @@ func preExecHook(cmd *cobra.Command, args []string) {
+		checkForUpdate()
+	}
+}
+
+func getRecentTimestamp() int64 {
+	potentialFiles, err := findFiles(os.TempDir())
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	var largestTimestamp int64 = 0
+	for i := 0; i < len(potentialFiles); i++ {
+		stringSections := strings.Split(potentialFiles[i].Name(), "/")
+		desiredSections := strings.Split(stringSections[len(stringSections)-1], "fsoctimestamp")
+		newTimestamp, _ := strconv.ParseInt(desiredSections[0], 10, 64)
+		if largestTimestamp < newTimestamp {
+			largestTimestamp = newTimestamp
+		}
+	}
+
+	for i := 0; i < len(potentialFiles); i++ {
+		err := os.Remove(potentialFiles[i].Name())
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+	}
+
+	return largestTimestamp
+}
+
+func findFiles(root string) ([]*os.File, error) {
+	var files []*os.File
+	err := filepath.WalkDir(root, func(pathh string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
+			if ok := strings.Contains(pathh, "fsoctimestamp"); ok && err == nil {
+				file, err := os.Open(pathh)
+				if err != nil {
+					log.Fatalf(err.Error())
+				}
+				files = append(files, file)
+			}
+		}
+		return nil
+	})
+	return files, err
+}
+
+func checkForUpdate() {
+	newestVersionChan := make(chan string)
+	go func() {
+		err := getVersion(newestVersionChan)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+	}()
+	var newestVersion = <-newestVersionChan
+	if compareVersion(newestVersion) {
+		log.Warnf("There is a newer version of FSOC available, please upgrade to version %s", newestVersion)
+	}
+}
+
+func getVersion(ver chan string) error {
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	resp, err := client.Get("https://github.com/cisco-open/fsoc/releases/latest")
+	if err != nil {
+		return err
+	}
+
+	split := strings.Split(resp.Header.Get("Location"), "/")
+	ver <- split[len(split)-1]
+	return nil
+}
+
+func compareVersion(newestVersion string) bool { // Returns if the parameter is a newer version than the installed one
+	currentVersion := version.GetVersion()
+	newestVersionMajor, newestVersionMinor, newestVersionPatch := parseVersion(newestVersion)
+	newerMajorVersion := uint(newestVersionMajor) > currentVersion.VersionMajor
+	newerMinorVersion := uint(newestVersionMinor) > currentVersion.VersionMinor
+	newerPatchVersion := uint(newestVersionPatch) > currentVersion.VersionPatch
+	if newerMajorVersion {
+		return true
+	}
+	if newerMinorVersion {
+		return true
+	}
+	if newerPatchVersion {
+		return true
+	}
+	return false
+}
+
+func parseVersion(version string) (uint64, uint64, uint64) {
+	version = version[1:]
+	versionSections := strings.Split(version, ".")
+	major, _ := strconv.ParseUint(versionSections[0], 10, 32)
+	minor, _ := strconv.ParseUint(versionSections[1], 10, 32)
+	patch, _ := strconv.ParseUint(versionSections[2], 10, 32)
+	return major, minor, patch
+}
+
+func postExecHook(cmd *cobra.Command, args []string) {
```

Comment:
Why is this here/on post. Ideally, we should set the timestamp as soon as we complete the check successfully (regardless  of whether there is a newer version or not; maybe even regardless of whether we got the latest version and it parsed OK)